packages/blockchain-wallet-v4/src/redux/common/bch/selectors.js

```diff
@@ -196,7 +196,7 @@ export const getNextAvailableChangeAddress = curry(
       HDWallet.selectAccount(accountIndex),
       walletSelectors.getDefaultHDWallet
     )(state)
-    const xpub = HDAccount.selectXpub(account)
+    const xpub = HDAccount.selectXpub(account, 'legacy')
```

Comment:
This is a separate issue that causes wallet to derive only the first address in the chain.